include/nccl-headers/tuner.h

```diff
@@ -0,0 +1,73 @@
+/*************************************************************************
+ * Copyright (c) 2023, NVIDIA CORPORATION. All rights reserved.
+ * Copyright (c) 2023, Meta Platforms, Inc. and affiliates.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+
+#ifndef NCCL_TUNER_H_
+#define NCCL_TUNER_H_
+
+#include "nccl-headers/net.h"
+#include "nccl-headers/error.h"
+
+#define NCCL_NUM_FUNCTIONS 5 // Send/Recv not included for now
+typedef enum { ncclFuncBroadcast, ncclFuncReduce, ncclFuncAllGather, ncclFuncReduceScatter, ncclFuncAllReduce, ncclFuncSendRecv, ncclFuncSend, ncclFuncRecv, ncclNumFuncs} ncclFunc_t;
+
+#define NCCL_NUM_ALGORITHMS 6 // Tree/Ring/CollNet*
```

Comment:
any chances same defines are used in different header files ?   like NCCL_ALGO_TREE is likely used by others.    wouldnt you prefer we call it NCCL_OFI_ALGO_TREE etc ?

Reply:
Tuner.h is directly coming from NCCL, and we'd want to use the same enums the way the API works.  

That said, redundancy exists, and it is coming from NCCL assuming a tuner plugin and a network plugin are independent whereas we want it to be part of the same library. We can refactor some of this to share definitions with the existing network plugin headers, but that makes upgrades across versions a little trickier instead of plopping in the header from each NCCL release. WIll take this up with the NCCL team to make it better.

Reply:
The header is in the wrong place; it should be in include/nccl-headers/nvidia/tuner.h, and it should be an exact copy of the tuner.h that is in upstream NCCL.  We learned the hard way that editing these files from upstream just leads to un-debuggable pain and suffering.